src/collective/ifttt/actions/ifttt.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+from concurrent.futures import ThreadPoolExecutor
+from OFS.SimpleItem import SimpleItem
+from plone.app.contentrules import PloneMessageFactory as _
+from plone.app.contentrules.actions import ActionAddForm
+from plone.app.contentrules.actions import ActionEditForm
+from plone.app.contentrules.browser.formhelper import ContentRuleFormWrapper
+from plone.contentrules.rule.interfaces import IExecutable
+from plone.contentrules.rule.interfaces import IRuleElementData
+from zope import schema
+from zope.component import adapter
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.schema.vocabulary import SimpleTerm
+from zope.schema.vocabulary import SimpleVocabulary
+
+import logging
+
+
+logger = logging.getLogger('collective.ifttt')
+
+payload_options = SimpleVocabulary([
+    SimpleTerm(value=u'Description', title=_(u'Description of action')),
+    SimpleTerm(value=u'Username', title=_(u'Username of Editor')),
+    SimpleTerm(
+        value=u'Event start date/time', title=_(u'Event Start Date/Time')
+    ),
```

Comment:
@Shriyanshagro Values should be something simple, like `description`, `username`, `start`. (Event start is usually saved in field "start".)

Simple term constructor actually accepts three keyword arguments:

```
SimpleTerm(value=True, token="true", title=_(u'Yes'))
```

* value is the value saved in the database
* token is the value used as HTML form input value and should be utf-8 serialized string (or something implicitly convertible for that)
* title is the value shown for user on HTML form

The shorter form only works when value can be implicitly converted to bytes string (utf-8 encoded string), like in

```
SimpleTerm(value="true", title=_(u'Yes'))
```

Reply:
@Shriyanshagro 

The best option would could be to define those value as constants

```python
PAYLOAD_DESCRIPTION = 'description'
PAYLOAD_USERNAME = 'username'
PAYLOAD_START = 'start'

payload_options = SimpleVocabulary([
    SimpleTerm(value=PAYLOAD_DESCRIPTION, title=_(u'Description of action')),
    SimpleTerm(value=PAYLOAD_USERNAME, title=_(u'Username of Editor')),
    SimpleTerm(value=PAYLOAD_START, title=_(u'Event Start Date/Time'))
])
```

This would work without token-keyword, because values are simply enough to also work as form input value tokens.

Constants are good, because you could later compare using those instead of inline strings prone to typing errors when looking up the values for them.